### ISSUE TITLE
Updates for v0.6?

### ISSUE THREAD

**Issue by cormullion** [JuliaImages/Images.jl#609]
Any thoughts about when a new release of Images.jl will start working on Julia version 0.6?

```

               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.0-pre.alpha.0 (2017-02-28 13:15 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin13.4.0

julia> using Images

WARNING: deprecated syntax "typealias AbstractGray{T} Color{T,1}" at /Users/me/.julia/v0.6/Images/src/Images.jl:38.
Use "AbstractGray{T} = Color{T,1}" instead.

WARNING: deprecated syntax "typealias TransparentRGB{C<:AbstractRGB,T} TransparentColor{C,T,4}" at /Users/me/.julia/v0.6/Images/src/Images.jl:39.
Use "TransparentRGB{C<:AbstractRGB,T} = TransparentColor{C,T,4}" instead.

WARNING: deprecated syntax "typealias TransparentGray{C<:AbstractGray,T} TransparentColor{C,T,2}" at /Users/me/.julia/v0.6/Images/src/Images.jl:40.
Use "TransparentGray{C<:AbstractGray,T} = TransparentColor{C,T,2}" instead.

WARNING: deprecated syntax "typealias NumberLike Union{Number,AbstractGray}" at /Users/me/.julia/v0.6/Images/src/Images.jl:41.
Use "const NumberLike = Union{Number,AbstractGray}" instead.

WARNING: deprecated syntax "typealias RealLike Union{Real,AbstractGray}" at /Users/me/.julia/v0.6/Images/src/Images.jl:42.
Use "const RealLike = Union{Real,AbstractGray}" instead.
INFO: Precompiling module Images.

WARNING: deprecated syntax "typealias AbstractGray{T} Color{T,1}" at /Users/me/.julia/v0.6/Images/src/Images.jl:38.
Use "AbstractGray{T} = Color{T,1}" instead.

WARNING: deprecated syntax "typealias TransparentRGB{C<:AbstractRGB,T} TransparentColor{C,T,4}" at /Users/me/.julia/v0.6/Images/src/Images.jl:39.
Use "TransparentRGB{C<:AbstractRGB,T} = TransparentColor{C,T,4}" instead.

WARNING: deprecated syntax "typealias TransparentGray{C<:AbstractGray,T} TransparentColor{C,T,2}" at /Users/me/.julia/v0.6/Images/src/Images.jl:40.
Use "TransparentGray{C<:AbstractGray,T} = TransparentColor{C,T,2}" instead.

WARNING: deprecated syntax "typealias NumberLike Union{Number,AbstractGray}" at /Users/me/.julia/v0.6/Images/src/Images.jl:41.
Use "const NumberLike = Union{Number,AbstractGray}" instead.

WARNING: deprecated syntax "typealias RealLike Union{Real,AbstractGray}" at /Users/me/.julia/v0.6/Images/src/Images.jl:42.
Use "const RealLike = Union{Real,AbstractGray}" instead.
WARNING: could not import Base.slice into Images
WARNING: could not import Base.sub into Images

WARNING: deprecated syntax "typealias FloatLike{T<:AbstractFloat} Union{T,Gray{T}}" at /Users/me/.julia/v0.6/ImageTransformations/src/ImageTransformations.jl:10.
Use "FloatLike{T<:AbstractFloat} = Union{T,Gray{T}}" instead.

WARNING: deprecated syntax "typealias FloatColorant{T<:AbstractFloat} Colorant{T}" at /Users/me/.julia/v0.6/ImageTransformations/src/ImageTransformations.jl:11.
Use "FloatColorant{T<:AbstractFloat} = Colorant{T}" instead.

WARNING: deprecated syntax "abstract MapInfo{T}" at /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:51.
Use "abstract type MapInfo{T} end" instead.

WARNING: deprecated syntax "abstract AbstractClamp{T}<:MapInfo{T}" at /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:139.
Use "abstract type AbstractClamp{T}<:MapInfo{T} end" instead.

WARNING: deprecated syntax "typealias GrayArray{T<:Union{Fractional,Bool}} Union{AbstractArray{T},AbstractArray{Gray{T}}}" at /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:588.
Use "GrayArray{T<:Union{Fractional,Bool}} = Union{AbstractArray{T},AbstractArray{Gray{T}}}" instead.

WARNING: deprecated syntax "inner constructor MapNone(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:62".
Use "MapNone{T}(...) where T" instead.

WARNING: deprecated syntax "inner constructor BitShift(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:114".
Use "BitShift{T,N}(...) where {T,N}" instead.

WARNING: deprecated syntax "inner constructor ClampMin(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:149".
Use "ClampMin{T,From}(...) where {T,From}" instead.

WARNING: deprecated syntax "inner constructor ClampMax(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:165".
Use "ClampMax{T,From}(...) where {T,From}" instead.

WARNING: deprecated syntax "inner constructor ClampMinMax(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:176".
Use "ClampMinMax{T,From}(...) where {T,From}" instead.

WARNING: deprecated syntax "inner constructor Clamp(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:199".
Use "Clamp{T}(...) where T" instead.

WARNING: deprecated syntax "inner constructor ScaleMinMax(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:265".
Use "ScaleMinMax{To,From,S}(...) where {To,From,S}" instead.

WARNING: deprecated syntax "inner constructor ScaleSigned(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:334".
Use "ScaleSigned{T,S}(...) where {T,S}" instead.

WARNING: deprecated syntax "inner constructor ScaleAutoMinMax(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:365".
Use "ScaleAutoMinMax{T}(...) where T" instead.

WARNING: deprecated syntax "inner constructor ScaleMinMaxNaN(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:385".
Use "ScaleMinMaxNaN{To,From,S}(...) where {To,From,S}" instead.

WARNING: deprecated syntax "inner constructor Clamp01NaN(...) around /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:397".
Use "Clamp01NaN{T}(...) where T" instead.
ERROR: LoadError: LoadError: type UnionAll has no field abstract
Stacktrace:
 [1] macro expansion at /Users/me/.julia/v0.6/Images/src/map-deprecated.jl:431 [inlined]
 [2] anonymous at ./<missing>:?
 [3] include_from_node1(::String) at ./loading.jl:539
 [4] include(::String) at ./sysimg.jl:14
 [5] include_from_node1(::String) at ./loading.jl:539
 [6] include(::String) at ./sysimg.jl:14
 [7] anonymous at ./<missing>:2
while loading /Users/me/.julia/v0.6/Images/src/map-deprecated.jl, in expression starting on line 429
while loading /Users/me/.julia/v0.6/Images/src/Images.jl, in expression starting on line 60
ERROR: Failed to precompile Images to /Users/me/.julia/lib/v0.6/Images.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:673
 [2] require(::Symbol) at ./loading.jl:460

julia> 
```

I'm using:

```
Julia Version 0.6.0-pre.alpha.0
Commit 29bc2ac (2017-02-28 13:15 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i5-2500S CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, sandybridge)
```

**Comment by kvmanohar22** [comment]
Not sure of new release, but` teh/0.6` branch works fine with julia 0.6. Refer [this](https://github.com/JuliaImages/Images.jl/issues/604).

**Comment by cormullion** [comment]
@kvmanohar22 Thanks. I'll wait. :)

**Comment by timholy** [comment]
Yeah, if I merged that branch to master now, you wouldn't get an error when you precompiled Images. That would allow some dependent packages to pass tests, which would be a good thing.

The flip side is that some very crucial functionality wouldn't work, and the errors would be very bizarre and hard to interpret. So I've decided not to merge `teh/0.6` into master until we can actually pass tests.

**Comment by cormullion** [comment]
Cool, no problem. Is this to do with StaticArrays? I read that they were only for smaller arrays < 100 elements. Anyway, look forward to stabler times!

**Comment by timholy** [comment]
>  Is this to do with StaticArrays?

Yes. ImageFiltering.jl (upon with Images depends) uses them to define kernels. Kernels are often quite small, especially separable kernels which may be as little as size 3 in each factor. This is a textbook use for StaticArrays.

The problem is that on julia 0.6, StaticArrays breaks functions like `zero`, see https://github.com/JuliaArrays/StaticArrays.jl/issues/109. I decided that was too bad to inflict on anyone, safer just to have Images fail to precompile.

**Comment by maxruby** [comment]
Hi Tim. I have working wrapper methods for conversion of JuliaImages arrays to `cv::Mat` but in order to test these methods fully I need `Images`working.  Do you have any ideas on the expected timeline for a stable version of `Images.jl` with v0.6?

**Comment by timholy** [comment]
It's all up to StaticArrays; I've gotten everything working on 0.6 up through ImageFiltering (ColorTypes, Colors, ColorVectorSpace, ImageCore, ImageAxes, ImageMetadata), so on my end I think it's fairly close. However, I should warn you that there could be nontrivial stuff to fix in ImageFiltering and Images; there was in every other package, e.g., ColorTypes alone took a couple of weeks with both @Keno and I working on it.

If you're desperate, I'm sure @andyferris would love help on https://github.com/JuliaArrays/StaticArrays.jl/issues/113 (it's a really big job, and he's tackling it solo AFAICT).

But do you really require 0.6 for OpenCV?

**Comment by maxruby** [comment]
Thanks for the update!  No I don't absolutely need v0.6 for OpenCV, _but_ it would be nice to stick with it since OpenCV itself is working fine with it ... 
I just checked out the `teh/0.6` branch from Images.jl and I can get what I need for now so hopefully I can finalize the new implementation of the wrapper methods.

**Comment by andyferris** [comment]
FYI I think *StaticArrays* is about 3/4 of the way done.

**Comment by timholy** [comment]
Closed by #614
